docs/source/metabase.py

```diff
@@ -0,0 +1,186 @@
+"""
+A Sphinx builder putting documentation on Metabase.
+
+"""
+
+import os
+import re
+from typing import Any, Dict, List
+
+import sphinx.writers.text
+import requests
+from docutils.core import publish_string
+from sphinx.application import Sphinx
+from sphinx.builders.dummy import DummyBuilder
+from sphinx.util import logging
+from sphinx.writers.text import TextTranslator, TextWriter
+
+logger = logging.getLogger(__name__)
+
+# Such a shame to depend on globals…
+sphinx.writers.text.MAXWIDTH = float('inf')
+
+
+class MetabaseBuilder(DummyBuilder):
+    """Sphinx builder used to deploy descriptions on Metabase."""
+    name = 'metabase'
+    default_translator_class = TextTranslator
+    tables = None
+    writer = None
+    session = None
+
+    def _get(self, path: str) -> Dict[str, Any]:
+        """Launch a GET requests on Metabase API."""
+        response = requests.get(f'{self.config.metabase_url}/{path}', headers=self.headers)
+        response.raise_for_status()
+        return response.json()
```

Comment:
Although the connection will eventually be closed with python deletes the response object, I think it's cleaner to use the response as a context manager so it's closed immediately. Like:
```
with requests.get(...) as response:
    response.raise_for_status()
    return response.json()
```

A further fix would be to open a requests session once so that a single connection is used to metabase. I don't know how many tables are involved, but I see a lot of `_get` and `_put` below in loops. It's actually pretty easy to use, see https://requests.readthedocs.io/en/master/user/advanced/#session-objects.

In this case, something like:
```
@property
def session(self):
    if not self._session:
        session = requests.Session()
        # Gather headers
        with session.post(...) as response:
            response.raise_for_status()
            session_key = response.json()['id']
        session.headers.update({
            'Content-Type': 'application/json',
            'X-Metabase-Session': session_key,
        })
        self._session = session
    return self._session
```
Then later using it is easy. You just use `self.session` instead of `requests` and you no longer need to specify the headers because they're set in the session object.
```
def _get(self, path):
    with self.session.get(...) as response:
        response.raise_for_status()
        return response.json()
```

Reply:
I didn’t know at all that it was possible with Requests… Thanks a lot! I’ll fix the code to use a session.

Reply:
> I don't know how many tables are involved, but I see a lot of `_get` and `_put` below in loops.

Indeed. It now takes only ~3 minutes instead of ~15, and the code is cleaner.

Reply:
Yay for HTTP connection sharing!